### ISSUE TITLE
Segfaults on PNGs converted from JP2 by ImageMagick 

### ISSUE THREAD

**Issue by jonsneyers** [pornel/dssim#23]
It's probably a color profile or gamma issue or something, if I convert to .pam first and then to .png (to make sure that the PNG does not contain any such metadata), it works fine.

E.g. comparing these two images causes a segfault:
![original](https://cloud.githubusercontent.com/assets/15010068/14176588/126c3854-f752-11e5-9e3c-32a70fa17e8f.png)
![lossy_jp2](https://cloud.githubusercontent.com/assets/15010068/14176599/2206decc-f752-11e5-9b50-39f1c56e83e6.png)

**Comment by pornel** [comment]
Do you have a backtrace? Are you compiling with `USE_COCOA=1`?

**Comment by jonsneyers** [comment]
If I compile a debug version of DSSIM, I get the following:
```
dssim: src/dssim.c:685: dssim_compare: Assertion `modified_image' failed.

**Comment by pornel** [comment]
Ah, it's poor error handling. It's probably rejecting gamma = 1

https://github.com/pornel/dssim/blob/61824b80ce5727080bc25616d470931c0a817bce/src/main.c#L168

**Comment by pornel** [comment]
Fixed in 574a1f912dfe19ad0b7b6df112f473806343cfa9

Note that your image is very suspicious - it says to have gamma 1.0, but is closer to original when interpreted as having srgb gamma, so the gamma chunk is most likely invalid.

**Comment by jonsneyers** [comment]
It is what I got when I converted the original to JP2 with ImageMagick and then back to PNG. So it looks like ImageMagick is inserting a bogus gamma chunk for some reason.
